framework/src/modules/interoperability/mainchain/store.ts

```diff
@@ -33,6 +44,78 @@ export class MainchainInteroperabilityStore extends BaseInteroperabilityStore {
 		return true;
 	}
 
+	public async forward(ccmForwardContext: CCMForwardContext): Promise<void> {
+		const {
+			ccm,
+			eventQueue,
+			logger,
+			networkIdentifier,
+			getAPIContext,
+			getStore,
+		} = ccmForwardContext;
+		const apiContext = getAPIContext();
+		const tokenAPI = this.interoperableModuleAPIs.get(MODULE_ID_TOKEN) as TokenAPI | undefined;
+		const beforeCCMSendContext = createCCMsgBeforeSendContext({
+			ccm,
+			eventQueue,
+			getAPIContext,
+			logger,
+			networkIdentifier,
+			getStore,
+			feeAddress: EMPTY_FEE_ADDRESS,
+		});
+
+		if (!tokenAPI) {
+			return;
+		}
+
+		const receivingChainIDAsStoreKey = getIDAsKeyForStore(ccm.receivingChainID);
+		let receivingChainAccount;
+		try {
+			receivingChainAccount = await this.getChainAccount(receivingChainIDAsStoreKey);
+		} catch {
+			return;
+		}
+
+		const isLive = await this.isLive(receivingChainIDAsStoreKey, Date.now());
+
+		if (receivingChainAccount.status === CHAIN_ACTIVE && isLive) {
+			const isTokenTransferred = tokenAPI.forwardMessageFee(apiContext, ccm);
+
+			if (!isTokenTransferred) {
```

Comment:
In the LIP it says to check "if the previous call fails or returns False". I am not sure how is `isTokenTransferred` implemented but maybe you also need to check if it throws an error.